### PR TITLE
Use buffer local versions of option variables.

### PR DESCRIPTION
**Pull request by chaoren**
This allows editing binary and regular files in the same session.

### CODE REVIEW

File: plugin/hexmode.vim [hunk 1]
```diff
@@ -101,29 +99,29 @@ if has("autocmd")
...
-            \  exe "set nomod" |
-            \  let &ma=oldma | let &ro=oldro |
-            \  unlet oldma | unlet oldro |
-            \  call winrestview(oldview) |
-            \  let &undolevels = &undolevels |
+            \  exe "setlocal nomod" |
+            \  let &l:ma=:oldma | let &l:ro=b:oldro |
+            \  unlet b:oldma | unlet b:oldro |
+            \  call winrestview(b:oldview) |
+            \  let &l:ul = &l:ul |
```
Comment (chaoren) [inline plugin/hexmode.vim:124]: This seems like a no-op... I wonder why it's there to begin with.

Comment (fidian) [reply plugin/hexmode.vim:124]: It seems that way, I completely agree, but it came along with documentation in #6.

**Comment by fidian** [comment]
Thanks for the update.  Seems to work for me.  I'm a little bit cautious about those readonly and modifiable flags so perhaps I will need to do some more testing.

File: plugin/hexmode.vim [hunk 2]
```diff
@@ -21,41 +21,39 @@ function ToggleHex()
...
     " and clear the read-only flag for now
-    let l:modified=&mod
-    let l:oldreadonly=&readonly
-    let &readonly=0
-    let l:oldmodifiable=&modifiable
-    let &modifiable=1
+    let l:modified = &l:modified
+    let l:oldreadonly = &l:readonly
+    let l:oldmodifiable = &l:modifiable
+    setlocal readonly
```
Comment (chaoren) [inline plugin/hexmode.vim:27]: Oh whoops. This should be `noreadonly`.

**Pull request merged**